3rdparty/jvm/global_scalac_plugins.lock

```diff
@@ -1 +1,12 @@
+# This lockfile was autogenerated by Pants. To regenerate, run:
+#
+#    ./pants generate-lockfiles
+#
+# --- BEGIN PANTS LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---
+# {
+#   "version": 1,
+#   "generated_with_requirements": []
+# }
+# --- END PANTS LOCKFILE METADATA ---
```

Comment:
Given that we control the lockfile format, how about storing this directly in TOML? Still keep the "was autogenerated" header as a comment though.

Reply:
Ack that the generic lockfile header code currently is set up to read JSON. I wonder if we can refactor it to, for example, take a pre-parsed `dict`.

I'd bias towards a more intuitive lockfile format over reuse of our JSON code.

Reply:
we decided previously that storing the metadata directly in code wasn't necessary right now

Reply:
As for a reason to keep them, we want to control where the metadata appears (at the top of the file), and we want to make it clear that editing that part of the file in particular will break things. The TOML library doesn't make it easy to add those comments programmatically

Reply:
Okay, that's fine with me. Thanks for explaining!